pkg/kubelet/cm/topologymanager/socketmask/socketmask.go

```diff
@@ -0,0 +1,154 @@
+            	maskHolder = buildMaskHolder(socketAffinityInt64)
+	}
+      	glog.Infof("[socketmask] MaskHolder : %v", maskHolder) 
+        glog.Infof("[socketmask] %v is passed into arrange function", socketMask)   
+        arrangedMask := arrangeMask(socketAffinityInt64)                   
+        newMask := getTopologyAffinity(arrangedMask, maskHolder)
+       	glog.Infof("[socketmask] New Mask after getTopologyAffinity (new mask) : %v ",newMask)
+        finalMaskValue := parseMask(newMask)
+       	glog.Infof("[socketmask] Mask []Int64 (finalMaskValue): %v", finalMaskValue)
+	maskHolder = newMask     
+        glog.Infof("[socketmask] New MaskHolder: %v", maskHolder)
+	finalSocketMask := SocketMask(finalMaskValue)
+	return finalSocketMask, maskHolder
+}
+
+func buildMaskHolder(mask [][]int64) []string {
+        var maskHolder []string
+        outerLen := len(mask)
+        var innerLen int = 0 
+        for i := 0; i < outerLen; i++ {
+                if innerLen < len(mask[i]) {
+                        innerLen = len(mask[i])
+                }
+        }
+        var buffer bytes.Buffer
+        var i, j int = 0, 0
+        for i = 0; i < outerLen; i++ {
+                for j = 0; j < innerLen; j++ {
+                        buffer.WriteString("1")
+                }
+                maskHolder = append(maskHolder, buffer.String())
+                buffer.Reset()
+        }
+        return maskHolder
+}
+
+func getTopologyAffinity(arrangedMask, maskHolder []string) []string {
+        var topologyTemp []string
+        for i:= 0; i < (len(maskHolder)); i++ {
+                for j:= 0; j < (len(arrangedMask)); j++ {
+                        tempStr := andOperation(maskHolder[i],arrangedMask[j])
+                        if strings.Contains(tempStr, "1") {
+                                topologyTemp = append(topologyTemp, tempStr )
+                        }
+                }
+        }
+        duplicates := map[string]bool{}
+        for v:= range topologyTemp {
+                duplicates[topologyTemp[v]] = true
+        }
+        // Place all keys from the map into a slice.
+        topologyResult := []string{}
+        for key, _ := range duplicates {
+                topologyResult = append(topologyResult, key)
+        }
+
+        return topologyResult
+}
+
+func parseMask(mask []string) []int64 {
+        var maskStr string
+        min := strings.Count(mask[0], "1")
+        var num, index int
+
+        for i := 0; i < len(mask); i++ {
+                num = strings.Count(mask[i], "1")
+                if num < min {
+                        min = num
+                        index = i
+                }
+                maskStr = mask[index]
+        }
+        var maskInt []int64
+        for _, char := range maskStr {
+                convertedStr, err := strconv.Atoi(string(char))
+                if err != nil {
+                        glog.Errorf("Could not convert string to int. Err: %v", err)
```

Comment:
errors should start with lowercase.
atoi returns errors that look like this:
```
strconv.Atoi: parsing "some-string": invalid syntax
```
so ideally this message should describe what this conversation is, e.g.:
```
glog.Errorf("could not convert mask character: %v", err)
```